tests/Fusonic/Linq/Test/MiscTest.php

```diff
@@ -36,19 +32,25 @@ public function testConcat_ReturnsConcatenatedElements()
         $this->assertEquals("d", $all[3]);
     }
 
-    public function testConcat_ThrowsArgumentExceptionIfNoTraversableArgument()
+    /**
+     * @expectedException InvalidArgumentException
```

Comment:
@voorproever Exception asserting via attributes can be problematic. For example see this explanation from xunit (a popular .net unit testing library)

> Long-term use of [ExpectedException] has uncovered various problems with it. First, it doesn’t specifically say which line of code should throw the exception, which allows subtle and difficult-to-track failures that show up as passing tests. Second, it doesn’t offer the opportunity to fully inspect details of the exception itself, since the handling is outside the normal code flow of the test. Assert.Throws allows you to test a specific set of code for throwing an exception, and returns the exception during success so you can write further asserts against the exception instance itself.

https://xunit.github.io/docs/comparisons.html#note1

See also: 

> If you are throwing a custom exception, then the chances of the same exception occurring in your code in the wrong place are minimal. However, things change when you are testing for system exceptions.

http://hadihariri.com/2008/10/17/testing-exceptions-with-xunit/